### PR TITLE
chore: add semantic model to mutators

### LINKED ISSUE #2720: [SPIKE] Investigate the cost of resolving semantic models for all syntax trees.
**Is your feature request related to a problem? Please describe.**
So far we have denied the need for semantic models as we assumed the cost of resolving the semantic models outweighs the value. However we are not sure what the exact cost is for the resolution.

**Describe the solution you'd like**
I'd like to investigate what the exact cost is for resolving the semantic models on a big solution. If the cost is low (a few milliseconds) we can implement this and use the semantic models in our mutators. This would greatly improve our capabilities as we are now often blocked by the unknown types in a mutator.

A good starting point could be the [visual studio performance profiler](https://learn.microsoft.com/en-us/visualstudio/profiling/beginners-guide-to-performance-profiling?view=vs-2022) or some other tool to record performance.

### PR DESCRIPTION
**Pull request by richardwerkman**
fixes #2720

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/MutationTest/FsharpMutationProcess.cs [hunk 1]
```diff
@@ -18,7 +19,7 @@ public class FsharpMutationProcess : IMutationProcess
         private readonly ILogger _logger;
         private readonly IFileSystem _fileSystem;
         private readonly StrykerOptions _options;
-        private readonly BaseMutantOrchestrator<FSharpList<SynModuleOrNamespace>> _orchestrator;
+        private readonly BaseMutantOrchestrator<FSharpList<SynModuleOrNamespace>, SemanticModel> _orchestrator;
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/MutationTest/FsharpMutationProcess.cs:22]: This is a Roslyn type and does not work for fsharp.

File: src/Stryker.Core/Stryker.Core/MutationTest/IMutationProcess.cs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+using Microsoft.CodeAnalysis;
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/MutationTest/IMutationProcess.cs:1]: Interface should be language agnostic

File: src/Stryker.Core/Stryker.Core/Mutants/FsharpMutantOrchestrator.cs [hunk 3]
```diff
@@ -11,7 +12,7 @@
 namespace Stryker.Core.Mutants
 {
     /// <inheritdoc/>
-    public class FsharpMutantOrchestrator : BaseMutantOrchestrator<FSharpList<SynModuleOrNamespace>>
+    public class FsharpMutantOrchestrator : BaseMutantOrchestrator<FSharpList<SynModuleOrNamespace>, SemanticModel>
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/Mutants/FsharpMutantOrchestrator.cs:15]: We shouldn't introduce Roslyn types to the fsharp orchestrator

Comment (richardwerkman) [reply src/Stryker.Core/Stryker.Core/Mutants/FsharpMutantOrchestrator.cs]: I've switched to object as a placeholder type (as fsharp doesn't yet use the semantic model)

**Pull request merged**